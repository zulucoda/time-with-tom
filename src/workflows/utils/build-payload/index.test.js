import buildPayload from '.';

describe('Workflow - Utils - Build Payload - Unit Test', () => {
  test('should get payload from each workflow and merge into payload object', async () => {
    const workflowA = () => ({
      component: () => 'some react component A',
      payload: (values) => ({ finalValA: 'valA1 and valA2' }),
      validate: (values) => ({}),
      initialValues: {
        valA1: '',
        valA2: '',
      },
    });
    const workflowB = () => ({
      component: () => 'some react component B',
      payload: (values) => ({ valB1: 'valB1', valB2: 'valB2' }),
      validate: (values) => ({}),
      initialValues: {
        valB1: '',
        valB2: '',
      },
    });

    const workflows = [workflowA(), workflowB()];
    const setup = buildPayload(workflows);
    const values = {
      valA1: 'valA1',
      valA2: 'valA2',
      valB1: 'valB1',
      valB2: 'valB2',
    };
    const act = await setup(values);
    expect(act).toMatchSnapshot({
      finalValA: 'valA1 and valA2',
      valB1: 'valB1',
      valB2: 'valB2',
    });
  });
});
