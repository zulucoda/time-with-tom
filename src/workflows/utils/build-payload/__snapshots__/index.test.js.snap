// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflow - Utils - Build Payload - Unit Test should ONLY get payload for active workflows 1`] = `
Object {
  "finalValA": "valA1 and valA2",
}
`;

exports[`Workflow - Utils - Build Payload - Unit Test should get payload from each workflow and merge into payload object 1`] = `
Object {
  "finalValA": "valA1 and valA2",
  "valB1": "valB1",
  "valB2": "valB2",
}
`;
